keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,618 @@
+type AllocatedDeviceStatus struct {
+    // Driver specifies the name of the DRA driver whose kubelet
+    // plugin should be invoked to process the allocation once the claim is
+    // needed on a node.
+    //
+    // Must be a DNS subdomain and should end with a DNS domain owned by the
+    // vendor of the driver.
+    //
+    // +required
+    Driver string `json:"driver" protobuf:"bytes,2,rep,name=driver"`
+
+    // This name together with the driver name and the device name field
+    // identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+    //
+    // Must not be longer than 253 characters and may contain one or more
+    // DNS sub-domains separated by slashes.
+    //
+    // +required
+    Pool string `json:"pool" protobuf:"bytes,3,rep,name=pool"`
+
+    // Device references one device instance via its name in the driver's
+    // resource pool. It must be a DNS label.
+    //
+    // +required
+    Device string `json:"device" protobuf:"bytes,4,rep,name=device"`
+
+    // Conditions contains the latest observation of the device's state.
+    // If the device has been configured according to the class and claim
+    // config references, the `Ready` condition should be True.
+    //
+    // +optional
+    // +listType=atomic
+    Conditions []metav1.Condition `json:"conditions" protobuf:"bytes,5,rep,name=conditions"`
+
+    // DeviceInfo contains arbitrary driver-specific data.
+    //
+    // +optional
+    // +listType=atomic
+    DeviceInfo []runtime.RawExtension `json:"deviceInfo,omitempty" protobuf:"bytes,6,rep,name=deviceInfo"`
+
+    // NetworkDeviceInfo contains network-related information specific to the device.
+    //
+    // +optional
+    NetworkDeviceInfo NetworkDeviceInfo `json:"networkDeviceInfo,omitempty" protobuf:"bytes,7,rep,name=networkDeviceInfo"`
+}
+
+// NetworkDeviceInfo provides network-related details for the allocated device.
+// This information may be filled by drivers or other components to configure
+// or identify the device within a network context.
+type NetworkDeviceInfo struct {
+    // Interface specifies the name of the network interface associated with
+    // the allocated device. This might be the name of a physical or virtual
+    // network interface.
+    //
+    // +optional
+    Interface string `json:"interface,omitempty" protobuf:"bytes,1,rep,name=interface"`
+
+    // NetworkAddresses lists the network addresses assigned to the device's network interface.
+    // This can include both IPv4 and IPv6 addresses.
+    //
+    // +optional
+    NetworkAddresses []NetworkAddress `json:"networkAddresses,omitempty" protobuf:"bytes,2,rep,name=networkAddresses"`
+
+    // Mac represents the MAC address of the device's network interface.
+    //
+    // +optional
+    Mac string `json:"mac,omitempty" protobuf:"bytes,3,rep,name=mac"`
+}
+
+// NetworkAddress provides a network address related details such as IP and Mask.
+type NetworkAddress struct {
```

Comment:
why the indirection?

Reply:
I don't have specific use case for it, just in the future if we want to have the broadcast, unicast address and the scope, then we could still do it via this struct. Otherwise we could just have a `type CIDR string` instead.

Reply:
we should follow the same conventions we follow elsewhere, if at all possible

Reply:
we use `string` and `[]strings` for ips and cidrs, we should have `Addresses []strings`
If there is need to add new fields they can be added to the original struct,